api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
 
-    result.address = result.address
+@groups.route('/groups/<int:group_id>/managers')
+@jwt_required
+def read_all_managers(group_id):
+    query = db.session.query(Manager).filter_by(group_id=group_id)
+    try:
+        managers = get_all_queried_entities(query, request.args)
+    except QueryArgumentError as e:
+        return jsonify(e.message), e.code
+    return jsonify(manager_schema.dump(managers, many=True))
 
-    return jsonify(meeting_schema.dump(result))
 
+@groups.route('/groups/<int:group_id>/managers/<int:person_id>')
+@jwt_required
+def read_one_manager(group_id, person_id):
+    manager = db.session.query(Manager).filter_by(group_id=group_id, person_id=person_id).first()
+    if manager is None:
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} does not exist"), 404
+    return jsonify(manager_schema.dump(manager))
 
-@groups.route('/meetings/<meeting_id>', methods=['PATCH'])
+
+@groups.route('/groups/<int:group_id>/managers/<int:person_id>', methods=['PATCH'])
 @jwt_required
-def update_meeting(meeting_id):
+def update_manager(group_id, person_id):
+    manager_schema = ManagerSchema(exclude=['group_id', 'person_id'])
     try:
-        valid_meeting = meeting_schema.load(request.json)
+        valid_attributes = manager_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
-
-    if meeting is None:
-        return jsonify(msg="Meeting not found"), 404
+    manager = db.session.query(Manager).filter_by(group_id=group_id, person_id=person_id).first()
+    if manager is None:
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} does not exist"), 404
 
-    for key, val in valid_meeting.items():
-        setattr(meeting, key, val)
+    for key, val in valid_attributes.items():
+        setattr(manager, key, val)
 
+    db.session.add(manager)
     db.session.commit()
-    return jsonify(meeting_schema.dump(meeting))
 
+    return jsonify(manager_schema.dump(manager)), 200
 
-@groups.route('/meetings/delete/<meeting_id>', methods=['DELETE'])
+
+@groups.route('/groups/<int:group_id>/managers/<int:person_id>', methods=['DELETE'])
 @jwt_required
-def delete_meeting(meeting_id):
-    # USE WITH CARE!!! --- WILL DELETE MEETING AND ALL ATTENDANCE TO THAT MEETING
-    meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
+def delete_manager(group_id, person_id):
+    manager = db.session.query(Manager).filter_by(group_id=group_id, person_id=person_id).first()
 
-    if meeting is None:
-        return jsonify(msg='Meeting not found'), 404
+    if manager is None:
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} does not exist"), 404
 
-    for member in meeting.members:
-        db.session.delete(member)
-    db.session.delete(meeting)
+    db.session.delete(manager)
```

Comment:
Although this is a "delete" operation, we very seldom want to delete data from the database. Instead, set the `active` field to `False`. Please create an issue to review delete endpoints the entire code base; there _may_ be some cases in which we really do want to delete, but let's be sure that we are using `active` flags anywhere that we don't want to lose data forever.

Reply:
Okay, I see what you are saying. So do we want to implement all the DELETE endpoints with an `inactivate` operation? If so, should we change all the `read_all` endpoints to return only entities that are active?
I'm thinking if we keep the current endpoints, we can still make a PATCH request with {'active': 'false'} in the UI when we want to "archive" some entity, and only make a DELETE request when we _really_ want to delete it.

Reply:
The default design should be to use an `active` flag; there may be some cases where we want to actually delete something, but I can't think of any right away. Maybe something like whether a group member has attended a meeting could actually be deleted, for example, if the person taking attendance just made a mistake. But this should be the exception.

I like your idea of using `PATCH` instead of `DELETE`. I've thought of `DELETE` operations with an `active` flag as kind of a "virtual delete," but the use of `PATCH` here does seem like it's more in the spirit of RESTful API design.